### PR TITLE
mgr/dashboard: improve device selection modal for creating OSDs

### PR DESCRIPTION
**Pull request by bk201**
Display a hint that user should select at least one filter that can be
translated into DriveGroup attributes.

The total capacity are going to be added is also displayed.

Fixes: https://tracker.ceph.com/issues/43166

Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

## How to test
Enable orchestrator backends or load dummy data:
```
cd /ceph/build
bin/ceph mgr module enable test_orchestrator
bin/ceph orch set backend test_orchestrator
wget https://raw.githubusercontent.com/bk201/ceph-resources/master/issues/40335/dummy.json -O /tmp/dummy.json
bin/ceph test_orchestrator load_data -i /tmp/dummy.json
```

## Screenshots
**Hints about required filters**
Screenshot_20200205_200239

**Dispaly total capacity to be added**
Screenshot_20200205_200302
Screenshot_20200205_201148

**Tooltips for Add buttons**
Screenshot_20200205_200407
Screenshot_20200205_200420
Screenshot_20200205_200518

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by bk201** [comment]
jenkins retest this please

**Comment by bk201** [comment]
jenkins test dashboard

**Comment by bk201** [comment]
> @bk201 my suggestion would be to put the hint about the required filters on top of the page (maybe in a colored message box? I think @p-se also used those for the smart data integration, IIRC).

That's actually my first design, here is the screenshot:
Screenshot_20200218_135835

**Comment by callithea** [comment]
> > @bk201 my suggestion would be to put the hint about the required filters on top of the page (maybe in a colored message box? I think @p-se also used those for the smart data integration, IIRC).
> 
> That's actually my first design, here is the screenshot:
> Screenshot_20200218_135835

I like this design better than the text on the bottom of the dialog, IMHO it's more of an eyecatcher than a text on the bottom of the page.
Just a small improvement to the displayed text:
"At least one of these filters must be selected _in order to proceed_: [...]"
(Not sure if it should be displayed as a warning, maybe a info box is sufficient? But let's wait for more feedback.)

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-devices-selection-modal/osd-devices-selection-modal.component.html [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
+        </cd-inventory-devices>
+        <div *ngIf="canSubmit; else filtersTip">
+          <p class="text-center">
+            <span i18n>Number of devices: {{ filteredDevices.length }}. Raw capacity:
+              {{ capacity | dimlessBinary }}.</span>
+          </p>
         </div>
+        <ng-template #filtersTip>
+          <p class="text-center">
+            <span i18n>At least one of these filters must be applied:</span>
```
Comment (callithea) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-devices-selection-modal/osd-devices-selection-modal.component.html]:
```suggestion
            <span i18n>At least one of these filters must be applied in order to proceed:</span>
```

**Comment by bk201** [comment]
@tspmelo Thanks, like this?
Screenshot_20200303_193808

**Comment by tspmelo** [comment]
looks good

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-devices-selection-modal/osd-devices-selection-modal.component.html (hunk 1 above)
Comment (bk201) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-devices-selection-modal/osd-devices-selection-modal.component.html]: Thanks, fixed.

**Comment by bk201** [comment]
jenkins test make check

**Comment by bk201** [comment]
jenkins test make check

**Comment by callithea** [comment]
jenkins test make check

**Comment by bk201** [comment]
@tspmelo Do you mind looking https://github.com/ceph/ceph/pull/33081/commits/9ded6287590b0f189238272200de93a2c7f72377 again? The commit is needed for hiding the title. Separate the feature as a commit in case backport is needed.

**Comment by callithea** [comment]
jenkins test make check

**Comment by callithea** [comment]
jenkins test make check

**Comment by bk201** [comment]
jenkins test make check

**Comment by bk201** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
lgtm

**Pull request merged**